### ISSUE TITLE
Output of prec_sparse() not positive definite

### ISSUE THREAD

**Issue by changwoo-lee** [equinor/GraphSPME#48]
Hello, 

Thanks for developing and sharing the GraphSPME package, I need a method for approximating dense precision matrix to sparse precision matrix given its sparsity pattern, and this is exactly what I was looking for. 

However, I'm afraid that the output of `prec_sparse` with `symmetrization = T`  option does not necessarily give a positive (semi)definite precision matrix, especially when the correlation is high. I want to use the output of prec_sparse as a precision matrix of multivariate normal, so it must be positive (semi)definite.

Here is a reproducible example, based on the example provided in "arp_dimension_study.R". I tested with R 4.2.1 and GraphSPME version 0.0.1 as of today (8/23/2022), and ts.extend version 0.1.1. 

```
library(ts.extend)
library(GraphSPME)
library(Matrix)
library(MASS)
library(dplyr)
library(ggplot2)
library(ggpubr)
fbnorm <- function(M) sqrt(sum(M^2))

set.seed(123)
#Set the parameters
n <- 100
p <- 5 # changed to AR(5)
AR    <- rep(0.99,p)/p  ## changed from 0.8 to 0.99 (high correlation), resulting AR(5) with coefficients 0.198, 0.198 ..., 0.198.

# check stationarity
abs(polyroot(c(1,-AR))) > 1

# Find population quantities
ntime <- 50
Sigma <- ARMA.var(ntime, ar=AR)
Prec <- solve(Sigma)
G <- bandSparse(ntime, ntime, 
                (-p):p, 
                sapply((-p):p, function(order) rep(1,ntime-abs(order)))
)
# Sample data
xtr <- ts.extend::rGARMA(n,ntime, ar=AR)

# Compute different estimates
S <- cov(xtr)
Prec_le <- prec_sparse(xtr, G, shrinkage=F, symmetrization = T) 
Prec_gspme <- prec_sparse(xtr, G, shrinkage = T, symmetrization = T)

all(eigen(Prec_le)$values >= 0) # not positive semidefinite

all(eigen(Prec_gspme)$values >= 0) # not positive semidefinite
```

Thank you so much,
Changwoo

**Comment by Blunde1** [comment]
Hello @changwoo-lee 
Thanks for using `GraphSPME` and finding and posting the issue! We too need the estimated precision in a multivariate normal, so we appreciate your post a lot.
I think I know what the issue is, and also have a fix in mind.
I will assign myself and thus putting this on my to-do list -- will keep you posted.
Thanks,
Berent

**Comment by Blunde1** [comment]
@changwoo-lee do you have the opportunity to test #49 and see if this resolves your problem?

**Comment by changwoo-lee** [comment]
Yes now I have a positive definite matrix output, thanks so much!

**Comment by Blunde1** [comment]
@changwoo-lee great! Will get my colleagues to look at it and hopefully merge this over/shortly after christmas/new years. 🙂

**Comment by Blunde1** [comment]
See https://github.com/equinor/GraphSPME/pull/50
